service/history/queues/executable.go

```diff
@@ -252,8 +252,9 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 	}
 
 	defer func() {
-		if !errors.Is(retErr, consts.ErrResourceExhaustedBusyWorkflow) {
-			// if err is due to workflow busy, do not take any latency related to this attempt into account
+		if !errors.Is(retErr, consts.ErrResourceExhaustedBusyWorkflow) &&
+			!errors.Is(retErr, consts.ErrResourceExhaustedAPSLimit) {
```

Comment:
if the APS limit error is return via an RPC call (though I understand this may not happen for now), then I think the check can't pass. Because the `consts.ErrResourceExhaustedAPSLimit` created by `serviceerror.NewResourceExhausted` doesn't have the status `st` field initialized if I remember correctly. but errors returned from grpc have that field set.

I think we need another check somewhere between L274 and L282

Reply:
I'm not sure I understand. Are you saying we need a stronger check here on L256 that checks cause, not just equivalence to a constant?

And on L274 - L282, I thought we wanted to include APS limit errors in the `task_errors_throttled` metric, so no change should be needed there?

Reply:
> Are you saying we need a stronger check here on L256 that checks cause, not just equivalence to a constant
Yes.

> on L274 - L282
I was thinking you can do something similar to L281 there, so that the new check you added on L425 can work. Otherwise the equivalence check there is also not enough. 
But I mean, you can certainly resolve those issues in different ways.

Reply:
Okay, with the most recent change we set the error on L279, before the deferred function is called, so I don't think L256 needs a stronger check, for the same reason BusyWorkflow doesn't.